public/app/features/alerting/unified/components/rules/deleted-rules/ConfirmRestoreDeletedRuleModal.tsx

```diff
@@ -0,0 +1,171 @@
+import { grafanaRuleDtoToFormValues } from '../../../utils/rule-form';
+import { rulerRuleType } from '../../../utils/rules';
+import { createRelativeUrl } from '../../../utils/url';
+
+type ModalProps = Pick<ComponentProps<typeof ConfirmModal>, 'isOpen' | 'onDismiss'> & {
+  isOpen: boolean;
+  ruleToRestore?: RulerRuleDTO | undefined;
+  onRestoreSucess: () => void;
+  onRestoreError: (error: Error) => void;
+};
+
+export const ConfirmRestoreDeletedRuleModal = ({
+  isOpen,
+  ruleToRestore,
+  onDismiss,
+  onRestoreSucess,
+  onRestoreError,
+}: ModalProps) => {
+  const [restoreMethod, { error }] = useRestoreDeletedRule();
+  const title = t('alerting.deletedRules.restore-modal.title', 'Restore deleted alert rule');
+  const errorTitle = t('alerting.deletedRules.restore-modal.error', 'Could not restore deleted alert rule');
+  const confirmText = !error
+    ? t('alerting.deletedRules.restore-modal.confirm', 'Yes, restore deleted rule')
+    : 'Manually restore the rule';
+
+  const styles = useStyles2(getStyles);
+
+  async function onRestoreConfirm() {
+    if (!ruleToRestore) {
+      return;
+    }
+    if (!rulerRuleType.grafana.rule(ruleToRestore)) {
+      return;
+    }
+    return restoreMethod
+      .execute(ruleToRestore)
+      .then(() => {
+        onDismiss();
+        onRestoreSucess();
+      })
+      .catch((err) => {
+        onRestoreError(err);
+      });
+  }
+
+  async function onManualRestore() {
+    if (!ruleToRestore) {
+      return;
+    }
+    if (!rulerRuleType.grafana.rule(ruleToRestore)) {
+      return;
+    }
+    const namespaceName = await backendSrv
+      .getFolderByUid(ruleToRestore.grafana_alert.namespace_uid)
+      .then((folder) => folder.title);
+    createAlert(ruleToRestore, namespaceName);
+  }
+
+  return (
+    <ConfirmModal
+      isOpen={isOpen}
+      title={title}
+      confirmText={confirmText}
+      modalClass={styles.modal}
+      confirmButtonVariant={!error ? 'destructive' : 'primary'}
+      body={
+        <Stack direction="column" gap={2}>
+          <Trans i18nKey="alerting.deletedRules.restore-modal.body">
+            Are you sure you want to restore this deleted alert rule definition?
+          </Trans>
+
+          <div>{ruleToRestore && <RulePeview rule={ruleToRestore} />}</div>
+          {error && (
+            <Alert severity="warning" title={errorTitle}>
+              <Trans i18nKey="alerting.alertVersionHistory.restore-deleted-manually">
+                Your alert rule could not be restored. This may be due to changes to other entities such as contact
+                points, data sources etc. Please manually restore the deleted rule.
+              </Trans>
+              <pre style={{ marginBottom: 0 }}>
+                <code>{stringifyErrorLike(error)}</code>
+              </pre>
+            </Alert>
+          )}
+        </Stack>
+      }
+      onConfirm={!error ? onRestoreConfirm : onManualRestore}
```

Comment:
This looks a bit confusing. Why we wan to invoke `onManualRestore` if there is an error?

Reply:
It looks like there are 2 options for restoring? One automatically restores the rule and the other goes through the form, am I correct?
I think it would be good to make it more explicit

Reply:
it only goes through the form if we get an error while restoring the first time.